### PR TITLE
fix issue 5020, allow ML.NET to load tf model with primitive input and output column

### PR DESCRIPTION
**Pull request by frank-dong-ms**
Fix issue 5020, enable tensorflow transformer take primitive type as input column.

### CODE REVIEW

File: src/Microsoft.ML.TensorFlow/TensorflowTransform.cs [hunk 1]
```diff
@@ -503,20 +503,20 @@ private sealed class Mapper : MapperBase
                         throw Host.Except("Variable length input columns not supported");
 
                     _isInputVector[i] = type is VectorDataViewType;
-                    if (!_isInputVector[i])
-                        throw Host.Except("Non-vector columns are not supported and should be loaded as vector columns of size 1");
-                    vecType = (VectorDataViewType)type;
+                    //if (!_isInputVector[i])
+                    //    throw Host.Except("Non-vector columns are not supported and should be loaded as vector columns of size 1");
```
Comment (harishsk) [inline src/Microsoft.ML.TensorFlow/TensorflowTransform.cs:507]: If this is not applicable anymore, please delete the lines.

File: src/Microsoft.ML.TensorFlow/TensorflowTransform.cs [hunk 2]
```diff
@@ -928,8 +926,8 @@ public SchemaShape GetOutputSchema(SchemaShape inputSchema)
                 var input = _options.InputColumns[i];
                 if (!inputSchema.TryFindColumn(input, out var col))
                     throw _host.ExceptSchemaMismatch(nameof(inputSchema), "input", input);
-                if (!(col.Kind == SchemaShape.Column.VectorKind.Vector))
-                    throw _host.ExceptSchemaMismatch(nameof(inputSchema), "input", input, "vector", col.GetTypeString());
+                //if (!(col.Kind == SchemaShape.Column.VectorKind.Vector))
```
Comment (harishsk) [inline src/Microsoft.ML.TensorFlow/TensorflowTransform.cs:929]: Please delete if not applicable.

**Comment by harishsk** [comment]
Please add a more descriptive title to the PR

**Review by harishsk — APPROVED** [review]
:shipit:

File: src/Microsoft.ML.TensorFlow/TensorTypeExtensions.cs [hunk 3]
```diff
@@ -14,6 +15,16 @@ internal static class TensorTypeExtensions
     {
         public static void ToScalar<T>(this Tensor tensor, ref T dst) where T : unmanaged
         {
+            //In ML.NET we are using ReadOnlyMemory<Char> to store string data but ReadOnlyMemory<Char>
+            //is not valid data type for tensorflow.net and exception will thrown if we call as_dtype method
+            //so we specially deal with string type here.
+            //Get string data first then convert to ReadOnlyMemory<Char> and assign value to dst.
+            if (typeof(T).Name.Contains("ReadOnlyMemory"))
```
Comment (eerhardt) [inline src/Microsoft.ML.TensorFlow/TensorTypeExtensions.cs:22]: Should this be `if (typeof(T) == typeof(ReadOnlyMemory<char>))`?

Comment (frank-dong-ms) [reply src/Microsoft.ML.TensorFlow/TensorTypeExtensions.cs]: Thanks, fixed

---
In reply to: [517666330](https://github.com/dotnet/machinelearning/pull/5468#discussion_r517666330) [](ancestors = 517666330)

**Comment by frank-dong-ms** [comment]
Thanks, done

---
In reply to: [721935322](https://github.com/dotnet/machinelearning/pull/5468#issuecomment-721935322) [](ancestors = 721935322)

**Pull request merged**